appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/ACCClassLoader.java

```diff
@@ -262,49 +234,40 @@ public ACCClassLoader run() {
 
     private byte[] readByteCode(final String className) throws ClassNotFoundException {
         final String resourceName = className.replace('.', '/') + ".class";
-        InputStream is = getResourceAsStream(resourceName);
-        if (is == null) {
-            throw new ClassNotFoundException(className);
-        }
-        try {
-            final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try (InputStream is = getResourceAsStream(resourceName);
+                ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+            if (is == null) {
+                throw new ClassNotFoundException(className);
+            }
             final byte[] buffer = new byte[8196];
             int bytesRead;
             while ( (bytesRead = is.read(buffer)) != -1) {
                 baos.write(buffer, 0, bytesRead);
```

Comment:
If we move beyond Java9 the whole block could be replace with a transfer found in JDK.